internal/compiler/object_tree.rs

```diff
@@ -993,27 +1097,63 @@ impl Element {
         diag: &mut BuildDiagnostics,
         tr: &TypeRegister,
     ) -> ElementRc {
+        let mut interfaces: Vec<Rc<Component>> = Vec::new();
         let base_type = if let Some(base_node) = node.QualifiedName() {
             let base = QualifiedTypeName::from_node(base_node.clone());
             let base_string = base.to_smolstr();
-            match parent_type.lookup_type_for_child_element(&base_string, tr) {
-                Ok(ElementType::Component(c)) if c.is_global() => {
+            match (
+                parent_type.lookup_type_for_child_element(&base_string, tr),
+                expected_relationship_to_parent(&node),
+            ) {
+                (Ok(ElementType::Component(c)), _) if c.is_global() => {
                     diag.push_error(
                         "Cannot create an instance of a global component".into(),
                         &base_node,
                     );
                     ElementType::Error
                 }
-                Ok(ty) => ty,
-                Err(err) => {
+                (Ok(ElementType::Component(c)), Some(ParentRelationship::Implements)) => {
+                    if !c.is_interface() {
+                        diag.push_error(
+                            format!("Cannot implement {}. It is not an interface", base_string),
+                            &base_node,
+                        );
+                        ElementType::Error
+                    } else if !diag.enable_experimental {
```

Comment:
(maybe this should be the first condition?)

Reply:
Done.